js/qmwp-search-correlations.js

```diff
@@ -26,208 +25,221 @@ quantimodoSearch.config(function ($httpProvider) {
+                            unit: measurement.variable.abbreviatedUnitName
+                        }],
+                        function (result) {
+                            console.log(result);
+                        });
                 }, function () {
                     console.debug('dismissed');
                 });
 
-            };
+            });
+        };
+
+        $scope.openVarSettingsModal = function (variable) {
 
-            $scope.addMeasurement = function (variable) {
-                console.log('Going to add measurement for variable: ' + variable);
-                QuantimodoSearchService.getVariableByName(variable, function (varDetails) {
-                    console.log(varDetails);
+            console.log('Going change setting for: ' + variable);
+            QuantimodoSearchService.getVariableByName(variable, function (varDetails) {
+                QuantimodoSearchService.getUnitsForVariableByName(varDetails.name, function (units) {
+
+                    console.log('Variable details:' + varDetails);
 
                     var modalInstance = $uibModal.open({
-                        templateUrl: qmwpPluginUrl + '/templates/search-correlations/add-measurement-modal.html',
-                        controller: 'addMeasurementModalInstanceController',
+                        templateUrl: qmwpPluginUrl + '/templates/search-correlations/variable-settings-modal.html',
+                        controller: 'varSettingsModalInstanceController',
                         resolve: {
                             variable: function () {
                                 return varDetails;
+                            },
+                            varUnits: function () {
+                                return units.data;
                             }
                         }
                     });
 
-                    modalInstance.result.then(function (measurement) {
-                        //confirmed
-                        QuantimodoSearchService.addMeasurement(
-                            [{
-                                measurements: [{
-                                    value: measurement.value,
-                                    timestamp: moment(new Date(measurement.date)).unix()
-                                }],
-                                name: measurement.variable.name,
-                                source: 'QuantiModo',
-                                category: measurement.variable.category,
-                                combinationOperation: measurement.variable.combinationOperation,
-                                unit: measurement.variable.abbreviatedUnitName
-                            }],
-                            function (result) {
-                                console.log(result);
-                            });
+                    modalInstance.result.then(function (variable) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['variable' is defined but never used.](https://www.codacy.com/app/m_3/QuantiModo-WordPress-Plugin/file/1247676643/issues/source?bid=2836188&fileBranchId=2937519#l220)